kubernetes/samples/scripts/domain-lifecycle/helper.sh

```diff
@@ -0,0 +1,606 @@
+  serverStartPolicyPatch=$(echo ${domainJson} | jq "${replacePolicyCmd}")
+  patchJson="{\"spec\": {\"managedServers\": "${serverStartPolicyPatch}"}}"
+  eval $__result="'${patchJson}'"
+}
+
+#
+# Function to get sorted list of servers in a cluster.
+# The sorted list is created in 'sortedByAlwaysServers' array.
+# $1 - Domain resource in json format
+# $2 - Name of server 
+# $3 - Name of cluster 
+# $4 - Indicates if policy of current server would be unset
+#
+function getSortedListOfServers {
+  local domainJson=$1
+  local serverName=$2
+  local clusterName=$3
+  local withPolicy=$4
+  local policy=""
+  local sortedServers=()
+  local otherServers=()
+
+  configMap=$(${kubernetesCli} get cm ${domainUid}-weblogic-domain-introspect-cm \
+    -n ${domainNamespace} -o json)
+  topology=$(echo "${configMap}" | jq '.data["topology.yaml"]')
+  jsonTopology=$(python -c \
+    'import sys, yaml, json; print json.dumps(yaml.safe_load('"${topology}"'), indent=4)')
+  clusterTopology=$(echo ${jsonTopology} | jq -r '.domain | .configuredClusters[] | select (.name == '\"${clusterName}\"')')
+  dynaCluster=$(echo ${clusterTopology} | jq .dynamicServersConfig)
+  if [ "${dynaCluster}" == "null" ]; then
+    # Cluster is a configured cluster, get server names
+    servers=($(echo ${clusterTopology} | jq -r .servers[].name))
+    # Sort server names in numero lexi order
+    IFS=$'\n' sortedServers=($(sort --version-sort <<<"${servers[*]}" ))
+    unset IFS
+    clusterSize=${#sortedServers[@]}
+  else 
+    # Cluster is a dynamic cluster, calculate server names
+    prefix=$(echo ${dynaCluster} | jq -r .serverNamePrefix)
+    clusterSize=$(echo ${dynaCluster} | jq .dynamicClusterSize) 
+    for (( i=1; i<=$clusterSize; i++ )); do
+      localServerName=${prefix}$i
+      sortedServers+=(${localServerName})
+    done
+  fi
+  # Create arrays of ALWAYS policy servers and other servers
+  sortedServersSize=${#sortedServers[@]}
+  if [ "${sortedServersSize}" -gt 0 ]; then
+    for localServerName in "${sortedServers[@]}"; do
+      getEffectivePolicy "${domainJson}" "${localServerName}" "${clusterName}" policy
+      # Update policy when server name matches current server and unsetting
+      if [[ "${withPolicy}" == "UNSET" && "${serverName}" == "${localServerName}" ]]; then
+        policy=UNSET
+      fi
+      if [ "${policy}" == "ALWAYS" ]; then
+        sortedByAlwaysServers+=(${localServerName})
+      else
+        otherServers+=(${localServerName})
+      fi
+    done
+  fi
+  
+  # append other servers to the list of servers with always policy
+  otherServersSize=${#otherServers[@]}
+  if [ "${otherServersSize}" -gt 0 ]; then
+    for otherServer in "${otherServers[@]}"; do
+      sortedByAlwaysServers+=($otherServer)
+    done
+  fi
+}
+
+#
+# Get replica count for a cluster
+# $1 - Domain resource in json format
+# $2 - Name of cluster 
+# $3 - Return value containing replica count
+#
+function getReplicaCount {
```

Comment:
If replicas is not set on the cluster, should this should return .spec.replica?  and if that is not set, should it should return 0?

Reply:
Changed.